flashlight/fl/autograd/Variable.h

```diff
@@ -114,6 +114,15 @@ class Variable {
    */
   Tensor& tensor() const;
 
+  /**
+   * Creates a copy of this variable, but detached from the computation graph.
+   * @param forceDeepCopy Forces a deep copy of the underlying tensor,
+   * circumventing any copy on write behaviour. If false, uses the underlying
+   * tensor's copy constructor behaviour, which is backend dependent.
+   * @return returns the cloned and detached variable.
+   */
+  Variable copy(bool forceDeepCopy = false) const;
```

Comment:
That makes sense. The primary reasoning for this was for convenience as it would reduce the amount of boilerplate conditional statements in modules that may want to explicitly perform a deep copy. Given users can still achieve deep copy behaviour via `Variable(x.tensor().copy(),x.isCalcGrad())`, its not technically adding any new functionality, just convenience.